### ISSUE TITLE
Simple detector blocks for diagnosing bottlenecks and loss of belt compression

### ISSUE THREAD

**Issue by Tamren** [Anuken/Mindustry-Suggestions#3444]
### Describe the content or mechanics you are proposing.

Detector blocks are green 2x2 blocks you can place on the ground, they are essentially self contained logic displays with a specialized function and can perform several tasks depending on how you set them. Detector blocks monitor belts on the 12 adjacent tiles around them (that includes diagonals) and will change from green to red if certain conditions are met. In order to function you must click on the block while it is green and select an operating mode:

- A compression detector will turn red if a gap appears on a fully loaded belt, which indicates that throughput has been lost because inputs upstream are failing to fully compress items onto the belt.
- A bottleneck detector will turn red if a fully loaded belt stop moving, even momentarily, which indicates that full throughput has been lost because of a bottleneck downstream.
- Clicking on a green block opens up a small menu that allows you to change the operating mode. 
- If you wish to log any faults that occur you can place a logic display next to one or more detector blocks. The logic display will automatically read the detectors and display their output as a number counting up from zero. If a large logic display is used it will additionally display a moving graph to give you an idea of when faults occur and how often.
- When multiple detectors are connected to the same logic display recorded faults will be grouped by material and graphs will display matching colours. "1 Titanium" would indicate a fault was detected on a belt carrying titanium, but if you need to know what specific belt it was you must use separate displays.
- Clicking on a red detector block will reset it back to green and reset the counter if you have that enabled.

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

I had a more complicated idea starting out but I pared it down to these two functions because they're what really matter when designing an effective system. Depending on what you are processing your factory will typically measure inputs and outputs in terms of whole compressed belts and designs will strive towards one of two goals. Either your design will attempt to to fully consume at least one (or more likely, all) of the input belts routed into the factory. Or your factory will attempt to compress it's output into one or more fully compressed belts.

Hence being able to detect bottlenecking in an input belt or failure to compress an output belt is something that players will always want to know, and usually the only factor necessary to determine if a factory design "works" or not. Detector blocks give players an easy button solution to do that without having to program a custom processor block and output that signal to a logical display. That is something that the existing blocks are capable of doing, but also something that is beyond the patience (or frankly intelligence) of most players who just want to have fun.

The most common use case I can think of is measuring very small losses in performance. Sometimes an input belt will back up, but it only happens for a split second and can be difficult for a player to detect by eye. Similarly output belts (especially those fed from the sides) can sometimes have gaps that are very small and are hard to see, especially when items are moving around corners or have been sped up by an overdrive module. Less commonly items on a fully compressed belt will glitch and create a visible gap in the line which disappears further downstream as the items "stretch" to fill in the space. In cases like these a measuring tool would be very helpful because I am often unable to determine if what I'm seeing is a problem with my factory design or just a graphical glitch.

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by itcannotbe** [comment]
This has a high chance of not getting implemented because:

1. It's not simple. Currently every conveyor tile is it's own block. So for this to be useful, every conveyor tile needs this. Further more, a moving conveyor can have 2 or 3 items on it while moving while the items are "compressed"

2. If you're worried about shoving the most on your conveyors, use plastanium conveyors.

3. I doubt that additional logic blocks are going to be created since, the new planet will not have logic. And the existing logic just exist because it's already there.

**Comment by Tamren** [comment]
Oh that's right. I included the logic display integration because I thought it would be a cool way for the player to add a logging function using existing game assets. But now that you mention it I just remembered that support for logic blocks in general is getting deprecated in v7.

If blocks being self-contained is an issue then I think I have a much simpler solution that doesn't require interaction between blocks at all. I'll post it later.
